practice/blackjack/hand.py

```diff
@@ -0,0 +1,162 @@
+"""Module containing the Hand class"""
+from card import Card
+
+
+class Hand:
+    """A class for representing a Hand of cards"""
+
+    def __init__(self, cards):
+        self.cards = cards
+
+    def __eq__(self, other):
+        return (
+            self.cards == other.cards
+            )
+
+    def __repr__(self):
+        return 'Hand({!r})'.format(
+            self.cards
+        )
+
+
+ACES = ([Card('A', 'C'), Card('A', 'D')], ([Card('A', 'H'), Card('A', 'C')]))
+EXAMPLE_CARD = Card('2', 'H')
+EXAMPLE_HAND = Hand([Card('Q', 'S'), Card('J', 'C')])
+
+
+def add_card_to_hand(hand, card):
+    """Adds a card to an existing hand.
+
+    >>> add_card_to_hand(EXAMPLE_HAND, EXAMPLE_CARD)
+    Hand([Card('Q', 'S'), Card('J', 'C'), Card('2', 'H')])
+    """
+    cards_for_hand = hand.cards + [card]
+    new_hand = Hand(cards_for_hand)
+    return new_hand
+
+
+def score_card(card):
+    """Calculates score for a single card.
+
+    >>> score_card(EXAMPLE_CARD)
+    2
+    >>> score_card(Card('K', 'S'))
+    10
+    >>> score_card(Card('A', 'C'))
+    11
+    """
+    if card.rank == 'A':
+        card_score = 11
+    elif card.rank == 'J' or card.rank == 'Q' or card.rank == 'K':
+        card_score = 10
+    else:
+        card_score = int(card.rank)
+    return card_score
+
+
+def score_hand(hand):
+    """Calculates the score for a given hand.
+
+    >>> score_hand(EXAMPLE_HAND)
+    20
+    >>> score_hand(Hand([Card('Q', 'S'), Card('J', 'C'), Card('5', 'C')]))
+    25
+    >>> score_hand(Hand([Card('Q', 'S'), Card('A', 'C'), Card('5', 'C')]))
+    16
+    >>> score_hand(Hand([Card('A', 'S'), Card('A', 'C'), Card('5', 'C')]))
+    17
+    >>> score_hand(Hand([Card('A', 'S'), Card('A', 'C'), Card('A', 'H')]))
+    13
+    >>> score_hand(Hand([Card('Q', 'S'), Card('A', 'S'), Card('A', 'C'), Card('A', 'H')]))
+    13
+    >>> score_hand(Hand([Card('Q', 'S'), Card('A', 'S'), Card('A', 'C'), Card('A', 'H'), Card('A', 'D')]))
+    14
+
+    """
+    scores = [score_card(card) for card in hand.cards]
+    temp_score = sum(scores)
+    if temp_score > 21 and (
+        Card('A', 'C') in hand.cards or
+        Card('A', 'D') in hand.cards or
+        Card('A', 'H') in hand.cards or
+        Card('A', 'S') in hand.cards
+    ):
```

Comment:
You basically want to know if an ace is in the hand. Another way of phrasing this is is any ace in the hand? Then you can use the `any()` function and explicitly pull out all of the ranks.

```py
if any([card.rank == 'A' for card in hand.cards]):
```
```